src/reports/orl/report.ts

```diff
@@ -134,30 +138,74 @@ export function generateORLReport(
     ...liabilitiesColumns.map((c) => r.liabilities[c.key] || 0),
   ]);
 
+
+    // Take receivables and liabilities from previous datasheet into account
+    // TODO: Improve this and handle the case when there's a difference between the number of columns (liabilitiesColumns).
+    if (previousLineNumber && previousDatasheet) {
+        const dataRangeNotation = previousDatasheet + "!A6:I" + previousLineNumber;
+        const oldDataRange = SpreadsheetApp.getActive().getRange(dataRangeNotation).getValues();
+        const rowLength = rows[0].length;
+        for (const oldRow of oldDataRange) {
+            const existingRow = rows.find(function (row) {
+                return String(row[0]).includes("/" + oldRow[0] + "/");
+            });
+            if (existingRow) {
+                existingRow[4] = Number(existingRow[4]) + Number(oldRow[4]);
+                existingRow[5] = Number(existingRow[5]) + Number(oldRow[5]);
+            } else {
+                const arr = new Array(rowLength - 6).fill(undefined);
+                rows.push([
+                    createHyperLinkForRecord(property, {
+                        liabilities: {total: 0},
+                        receivables: 0,
+                        transactions: [],
+                        id: oldRow[0],
+                        type: oldRow[3] === "External" ? "External" : "Guest"
+                    }),
+                    oldRow[1],
+                    oldRow[2],
+                    oldRow[3],
+                    oldRow[4],
+                    oldRow[5],
+                    ...arr
+                ]);
+            }
+            totals.receivables += oldRow[4];
+            totals.liabilities.total += oldRow[5];
```

Comment:
to avoid this naming like `old`, we can create a function which return `receivables` and `total` for `data range` parameter